pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -464,7 +464,12 @@ stdenv.mkDerivation ({
   installPhase = ''
     runHook preInstall
 
-    ${if !isLibrary then "${setupCommand} install" else ''
+    ${if !isLibrary && buildTarget == "" then "${setupCommand} install"
```

Comment:
@cdepillabout This line is what should preserve existing packages without rebuilds.